apps/manager_app/get_device_info.c

```diff
@@ -0,0 +1,150 @@
+ *
+ *
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License,
+ * as defined below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of
+ * rights under the License will not include, and the License does not
+ * grant to you, the right to Sell the Software.
+ *
+ * For purposes of the foregoing, "Sell" means practicing any or all
+ * of the rights granted to you under the License to provide to third
+ * parties, for a fee or other consideration (including without
+ * limitation fees for hosting or consulting/ support services related
+ * to the Software), a product or service whose value derives, entirely
+ * or substantially, from the functionality of the Software. Any license
+ * notice or attribution required by the License must also include
+ * this Commons Clause License Condition notice.
+ *
+ * Software: All X1Wallet associated files.
+ * License: MIT
+ * Licensor: HODL TECH PTE LTD
+ *
+ ******************************************************************************
+ */
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include "controller_level_four.h"
+#include "flash_api.h"
+#include "manager_app.h"
+
+/*****************************************************************************
+ * EXTERN VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Returns the formatted semantic versioning.
+ *
+ * @param firmware_version
+ * @return bool Status indicating the
+ */
+bool get_firmware_version(common_version_t *firmware_version);
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+bool get_firmware_version(common_version_t *firmware_version) {
```

Comment:
How about we refactor the `get_fwVer()` API directly to return `common_version_t` instead of making more and more wrapper functions?

All other calls needs to be updated anyway

Just a suggestion, we can choose to keep it this way only

Reply:
If this was a custom structure, then it would have been okay, but `common_version_t` is a nanopb-generated structure, when including this into flash_api, we would be creating unnecessary dependency in that file.

If this were a custom-defined structure, then the suggestion is valid.